10-deploying-an-application/modules/my_app/plans/deploy.pp

```diff
@@ -0,0 +1,95 @@
+plan my_app::deploy(
+  Pattern[/\d+\.\d+\.\d+/] $version,
+  TargetSpec $app_servers,
+  TargetSpec $db_server,
+  TargetSpec $lb_server,
+  String[1] $instance = 'my_app',
+  Boolean $force = false
+) {
+  # Validate that there is only a single lb server to check
+  if get_targets($lb_server).length > 1 {
+    fail_plan("${lb_server} did not resolve to a single target")
+  }
+
+  # First query the load balancer and make sure the app isn't under too much load to do a deploy.
+  unless $force {
+    $conns = run_task('my_app::lb', $lb_server,
+       "Check load before starting deploy",
+       action => 'stats',
+       backend => $instance,
+       server => 'FRONTEND',
+    ).first['connections']
+    if ($conns > 8) {
+      fail_plan("The appplication has too many open connections: ${conns}")
+    } else {
+      # Info messages will be displayed when the --verbose flag is used.
+      info("Application has ${conns} open connections.")
+    }
+  }
+
+  # Install the new version of the application and check what version was previously
+  # installed so it can be deleted after the deploy.
+  $old_versions = run_task('my_app::install', [$app_servers, $db_server],
+    "Install ${version} of the application",
+    version => $version
+  ).map |$r| { $r['previous_version'] }
+
+  run_task('my_app::migrate', $db_server)
+
+  # Don't log every action on each node only important messages
+  without_default_logging || {
```

Comment:
I tried adding this to my path, and got `Syntax error at '|' (file: /home/lucy/githubs/tasks-hands-on-lab/10-deploying-an-application/modules/my_app/plans/deploy.pp, line: 40, column: 27)\n   (file & line not available)`. 

```
$ ruby --version
ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]
```

Reply:
yeah it needs a ()